package/gluon-mesh-vpn-wireguard/README.md

```diff
@@ -0,0 +1,40 @@
+# gluon-mesh-vpn-wireguard
+
+This package will allow wireguard [1] to be used in gluon. Wireguard being a
+VPN on OSI layer 3 is useful for mesh protocols that operate on layer 3.
+
+[1] https://wireguard.io
+
+## including servers via site.conf
+This is similar to the fastd-based mesh_vpn structure.
+
+```
+          mesh_vpn = {
+                  mtu = 1374,
+
+                  wireguard = {
+                          configurable = '1',
+                          enabled = '1',
+                          groups = {
+                                  backbone = {
+                                          limit = '2', -- currently unused
+                                          peers = {
+                                                  gw02 = {
+                                                          iface = 'wg-mesh-gw02',
+                                                          enabled = '1',
+                                                          PublicKey ='bog2DzyiC0Os7y1GloEw0afb8bLdZ9SzVQCd44Eock4=',
+                                                          remote = 'gw02.babel.ffm.freifunk.net:40100',
```

Comment:
Does not work, because port is already hardcoded in `wg-broker`

Reply:
The broker need a port (like in line below) - but does the `remote` need it? (it should be set by broker)

Reply:
`s/remote/broker/` should do it imo. The client doesn't need to know anything about the server and should get all it needs from the broker.

I don't like your suggestion regarding `s/PublicKey/key/` because `key` is usually used for shared keys and not keypairs. An interface with a single peer has:
- A private key.
- A public key.
- A shared key (optional).

Which one is meant by an attribute called `key`?

Reply:
Okay, but this should be also at there sourcecode ...
Take a look here `lib/gluon/upgrade/405-wireguard` on `add_peer` function:
https://github.com/freifunk-gluon/gluon/pull/1534/files#diff-f73520f2291e9f4839e8822ff3a022d0R41